### ISSUE TITLE
Panel fabric A11y Pass

### ISSUE THREAD

**Issue by awalia13** [OfficeDev/office-ui-fabric-react#8173]
Tracking all of the issues brought up with the 03/01 accessibility pass on Panel fabric control
Details on individual bugs can be found on [Sharepoint](https://microsoft-my.sharepoint.com/:x:/r/personal/v-amaver_microsoft_com/_layouts/15/Doc.aspx?sourcedoc=%7B5afd015d-fc38-4e4b-ae9b-a0ad778c0ea7%7D&action=default&uid=%7B5AFD015D-FC38-4E4B-AE9B-A0AD778C0EA7%7D&ListItemId=1850&ListId=%7B395E1ECB-5002-440E-81C5-6A090E98077C%7D&odsp=1&env=prodbubble
)
- [ ] Bug1: [Accessibility][Programmatic access]: Name is not defined for close button in the ‘Panel - Small, left-aligned, fixed’ dialog.
- [ ] Bug2: [Accessibility][Programmatic access]: Name is not defined for see more(3 dots) button in the Panel with custom navigation content dialog.
- [ ] Bug3: [Accessibility][Usable]: In scan mode focus goes twice on close button.
- [ ] Bug4: [Accessibility][Usable]: In scan mode after navigating to the last item on ‘Panel’ dialog instead of saying no next item focus keeps on rotating on last and second last element.
- [ ] Bug5: [Accessibility][Usable]:Narrator narrates irrelevant information on expanding the see more(3 dots) button in ‘Panel with custom navigation content’ dialog.
- [ ] Bug6: [Accessibility][Keyboard Navigation]: Scroller which is present in 'Panel with scrolling content' dialog is not accessible by keyboard.
- [ ] Bug7: [Accessibility][Forms and validation]: Label is not associated with the text field in ‘Panel’ dialog.
- [ ] Bug8: [Accessibility][Visual Requirements] Tooltip is not present for see more(3 dots) button in ‘Panel with custom navigation content’ dialog.
- [ ] Bug9: [Accessibility][Supporting the Platform]: After zooming the browser to 200% Scroller appears which is not accessible by keyboard.
- [ ] Bug10: [Accessibility][Programmatic access]: ARIA attributes must conform to valid values
- [ ] Bug11: [Accessibility][Usable]: Scan mode focus goes separately on the menu item and its icon in see more(3 dots) drop down.
- [ ] Bug12: [Accessibility][Usable][Suggestion]: Heading in ‘Panel with custom navigation content’ dialog should be above search box.  
- [ ] Bug13: [Accessibility][Usable]: Screen does not scroll down along with keyboard focus in Backward navigation. 
- [ ] Bug14: [Accessibility][Usable]: Screen reader is not reading the entire content of dialog box  when the dialog opens.
- [ ] Bug15: [Accessibility][Usable]: Focus moves out of the dialog(to the Microsoft logo) upon pressing CTRL+Home instead of moving to the first element of the dialog.

**Comment by aditima** [comment]
Moving to the Accessibility project backlog. We'll need to review and assign a milestone for this work.

**Comment by dcowen91** [comment]
Hi, Just curious if this issue has been scheduled? There are a ton of issues found by the [fastpass tool](https://accessibilityinsights.io/) even just opening a panel on the [fabric demo site](https://developer.microsoft.com/en-us/fabric#/components/panel), but I'm not sure if they are all enumerated here.

**Comment by jdhuntington** [comment]
1 - Fixed

**Comment by jdhuntington** [comment]
2 - Fixed

**Comment by jdhuntington** [comment]
3 - Unable to repro

**Comment by jdhuntington** [comment]
4 - Unable to repro (may have been fixed with focus zone fixes)

**Comment by jdhuntington** [comment]
5 - Fixed

**Comment by jdhuntington** [comment]
6 - Fixed

**Comment by jdhuntington** [comment]
7 - Fixed

**Comment by jdhuntington** [comment]
8 - Fixed

**Comment by jdhuntington** [comment]
9 - This isn't something that the panel can support for every use-case unfortunately. In this case, the scrollbar is accessible once an item inside gains focus.

**Comment by jdhuntington** [comment]
10 - `aria-describedby` no longer appears in the examples. However, it's not clear from the bug why `id__27` is invalid.

**Comment by jdhuntington** [comment]
11 - Fixed

**Comment by jdhuntington** [comment]
12 - Good suggestion - we likely need a better example here.

**Comment by jdhuntington** [comment]
13,14,15 - Will fix in other contexts.

**Comment by 999-Dev** [comment]
New issue found :

Bug16 : [Accessibility][Tracking of focus : Panel control] : Screen reader focus does not moves to the 'Close' button while backward navigation in scan mode.

Test Environment: 
OS: Version: RS5_Prerelease(1903 OS Build 18334.1) 
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/dialog
Edge Browser Version: 
Microsoft Edge: 44.18983.1.0
Microsoft EdgeHTML: 18.18983
Chrome Browser Version:
Chrome Version: 76.0.3809.132

**Comment by 999-Dev** [comment]
New issue found :
Bug17 : [Functional : Panel Control] : No action is getting preformed after activating the 'Close' button either by using keyboard or by mouse.

Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/dialog
Edge Browser Version:
Microsoft Edge: 44.18983.1.0
Microsoft EdgeHTML: 18.18983
Chrome Browser Version:
Chrome Version: 76.0.3809.132

**Comment by 999-Dev** [comment]
Bug1: [Accessibility][Programmatic access]: Name is not defined for close button in the ‘Panel - Small, left-aligned, fixed’ dialog.
Issue is fixed. Verified in the below build:
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug2: [Accessibility][Programmatic access]: Name is not defined for see more(3 dots) button in the Panel with custom navigation content dialog.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug3: [Accessibility][Usable]: In scan mode focus goes twice on close button.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug4: [Accessibility][Usable]: In scan mode after navigating to the last item on ‘Panel’ dialog instead of saying no next item focus keeps on rotating on last and second last element.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug9: [Accessibility][Supporting the Platform]: After zooming the browser to 200% Scroller appears which is not accessible by keyboard.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0
 
Bug10: [Accessibility][Programmatic access]: ARIA attributes must conform to valid values.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0
 
Bug11: [Accessibility][Usable]: Scan mode focus goes separately on the menu item and its icon in see more(3 dots) drop down.
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

**Comment by 999-Dev** [comment]
Bug5: [Accessibility][Usable]:Narrator narrates irrelevant information on expanding the see more(3 dots) button in ‘Panel with custom navigation content’ dialog.
Issue is not fixed. Verified in the below build:
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug6: [Accessibility][Keyboard Navigation]: Scroller which is present in 'Panel with scrolling content' dialog is not accessible by keyboard.
Issue is not fixed. Verified in the below build:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug7: [Accessibility][Forms and validation]: Label is not associated with the text field in ‘Panel’ dialog.
Issue is not fixed. Verified in the below build:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

Bug8: [Accessibility][Visual Requirements] Tooltip is not present for see more(3 dots) button in ‘Panel with custom navigation content’ dialog.
Issue is not fixed. Verified in the below build:
OS: Version: RS5_Prerelease(1903 OS Build 18334.1)
URL: https://fabricweb.z5.web.core.windows.net/oufr/7.35.0/#/examples/panel
Edge Browser Version:
Microsoft Edge: 44.18983.1.0

**Comment by 999-Dev** [comment]
Bug8: [Accessibility][Visual Requirements] Tooltip is not present for see more(3 dots) button in ‘Panel with custom navigation content’ dialog.
UI has been changed. See more button is not visible now. Verified in the below build:
OS: Version: RS5_Prerelease(1903 OS Build 18999.1)
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/panel
Edge Browser Version:
Microsoft Edge: 44.18999
Hence closing the bug.

**Comment by 999-Dev** [comment]
Bug5: [Accessibility][Usable]:Narrator narrates irrelevant information on expanding the see more(3 dots) button in ‘Panel with custom navigation content’ dialog.
UI has been changed. See more button is not visible now. Verified in the below build:
Issue is not fixed. Verified in the below build:
Test Environment:
OS: Version: RS5_Prerelease(1903 OS Build 18999.1)
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/panel
Edge Browser Version:
Microsoft Edge: 44.18999
Hence closing the bug.

**Comment by 999-Dev** [comment]
Bug7: [Accessibility][Forms and validation]: Label is not associated with the text field in ‘Panel’ dialog.
UI has been changed. See more button is not visible now. Verified in the below build: 
**Test Environment:**
OS: Version: RS5_Prerelease(1903 OS Build 18999.1)
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/panel
Edge Browser Version:
Microsoft Edge: 44.18999.1.0
Hence closing the bug.

**Comment by 999-Dev** [comment]
Bug6: [Accessibility][Keyboard Navigation]: Scroller which is present in 'Panel with scrolling content' dialog is not accessible by keyboard.
UI has been changed. See more button is not visible now. Verified in the below build:
**Test Environment:**
OS: Version: RS5_Prerelease(1903 OS Build 18999.1)
URL: https://developer.microsoft.com/en-us/fabric#/controls/web/panel
Edge Browser Version:
Microsoft Edge: 44.18999.1.0
Hence closing the bug.
